x11rb/src/rust_connection/stream.rs

```diff
@@ -352,81 +382,60 @@ fn do_write(
 
 impl Stream for DefaultStream {
     fn poll(&self, mode: PollMode) -> Result<()> {
-        #[cfg(unix)]
-        {
-            use nix::poll::{poll, PollFd, PollFlags};
+        use rustix::event::{poll, PollFd, PollFlags};
+        use rustix::io::Errno;
 
-            let mut poll_flags = PollFlags::empty();
-            if mode.readable() {
-                poll_flags |= PollFlags::POLLIN;
-            }
-            if mode.writable() {
-                poll_flags |= PollFlags::POLLOUT;
-            }
-            let fd = self.as_raw_fd();
-            let mut poll_fds = [PollFd::new(fd, poll_flags)];
-            loop {
-                match poll(&mut poll_fds, -1) {
-                    Ok(_) => break,
-                    Err(nix::Error::EINTR) => {}
-                    Err(e) => return Err(e.into()),
-                }
-            }
-            // Let the errors (POLLERR) be handled when trying to read or write.
-            Ok(())
+        let mut poll_flags = PollFlags::empty();
+        if mode.readable() {
+            poll_flags |= PollFlags::IN;
         }
-        #[cfg(windows)]
```

Comment:
It's nice that this complication goes away. I'm not quite sure how this magic works, but I guess CI ensures that it does work. Thanks! :-)